sqllogictest/src/parser.rs

```diff
@@ -982,6 +1016,59 @@ fn parse_multiline_error<'a>(
     ExpectedError::Multiline(parse_multiple_result(lines))
 }
 
+/// Parse retry configuration from tokens
+///
+/// The retry configuration is optional and can be specified as:
+///
+/// ```text
+/// ... retry 3 backoff 1s
+/// ```
+fn parse_retry_config(tokens: &[&str]) -> Result<Option<RetryConfig>, ParseErrorKind> {
+    if tokens.is_empty() {
+        return Ok(None);
+    }
+
+    let mut iter = tokens.iter().peekable();
+
+    // Check if we have retry clause
+    match iter.next() {
+        Some(&"retry") => {}
+        Some(token) => return Err(ParseErrorKind::UnexpectedToken(token.to_string())),
+        None => return Ok(None),
+    }
+
+    // Parse number of attempts
+    let attempts = match iter.next() {
+        Some(attempts_str) => attempts_str
+            .parse::<usize>()
+            .map_err(|_| ParseErrorKind::InvalidNumber(attempts_str.to_string()))?,
+        None => return Err(ParseErrorKind::UnexpectedEOF),
```

Comment:
Should this be a new variant like `InvalidRetryConfig` instead? It's not exactly end of _file_.